src/output.js

```diff
@@ -0,0 +1,96 @@
+const { IncomingWebhook } = require('@slack/webhook')
+const { readFileSync } = require('fs')
+
+/**
+ * @param {{type: string, status: number, slackWebhookUrl: string}} params
+ */
+async function run({ type, ...args }) {
+  try {
+    if (type === 'github') {
+      // tbd
+      // await githubCheck(args)
+    } else if (type === 'slack') {
+      await slackNotification(args)
+    }
+  } catch (e) {
+    throw e
+  }
+}
+
+/**
+ * @param {{status: number, slackWebhookUrl: string}} params
+ * @return {Promise<*>}
+ */
+async function slackNotification({ status, slackWebhookUrl }) {
+  const conclusion = status === 0 ? 'success' : 'failure'
+  const webhook = new IncomingWebhook(slackWebhookUrl)
+  const color = status === 0 ? 'good' : 'danger'
+
+  /** @type {Buffer} */
+  const assertionResults = readFileSync(`${process.cwd()}/.lighthouseci/assertion-results.json`)
+  const groupedResults = groupBy(JSON.parse(assertionResults.toString()), 'url')
+
+  const attachments = Object.values(groupedResults).map(groupedResult => {
+    const results = groupedResult.map(
+      /**
+       * @param {{auditId: string, auditProperty: string, auditTitle: string, expected: string, operator: string, actual: string, url: string}} res
+       * @return {{title?: string, type: string, value: string, url: string}}
+       */
+      res => ({
+        title: `${res.auditId}.${res.auditProperty}`,
+        value: `${res.auditTitle} \n _Expected ${res.expected} ${
+          res.operator === '<=' ? ' less then' : ' mode than'
+        } actual ${res.actual}_`,
+        type: 'mrkdwn',
+        url: res.url
+      })
+    )
+
+    const fields = results.slice(0, 2)
+    fields.length > 0 &&
+      fields.push({
+        title: '...'
+      })
+
+    return {
+      text: `${groupedResult.length + 1} result(s) for ${groupedResult[0].url}`,
+      color,
+      fields
+    }
+  })
+
+  return webhook.send({
+    attachments: [
+      {
+        pretext: 'GitHub Actions / LH Report',
+        title: `Pull Request ${conclusion} - <https://github.com/treosh/lighthouse-ci-action/|View on GitHub>`,
```

Comment:
Does this link to the Github action, or the actual invocation for the run this post is about?

Reply:
Sorry for the confusion. I put it as placeholder and forgot to put todo comment. What we wanna achieve here are:
1. link to PR so users can easy open PR and review changes
2. link to actual details (aka in github check)

(2) I'll comment it, it will be fixed with github checks PR, they will have kinda the same link

regarding (1)
@exterkamp @alekseykulikov can we use some env variable to get PR url.
I've looked through docs - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables
but not sure there is one for that.

Reply:
IIRC you can get that info from the octokit client. In some fashion similar to how Paul made his [suite checks](https://github.com/paulirish/lighthouse-ci-action/blob/checksuite/src/github.js#L9). You can list [pulls with the ocktokit client](https://octokit.github.io/rest.js/#octokit-routes-pulls) and probably some env from the action + that will give you the exact PR you want. You can also get data about specific checks from that [too afaik](https://octokit.github.io/rest.js/#octokit-routes-checks-list-for-ref)?